internal/xproto/patch.go

```diff
@@ -0,0 +1,124 @@
+package xproto
+
+import (
+	"github.com/pkg/errors"
+	"github.com/tableauio/tableau/proto/tableaupb"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+// PatchMessage patches src into dst, which must be a message with the same descriptor.
+//
+// # Patch option "PATCH_MERGE"
+//   - scalar: Populated scalar fields in src are copied to dst.
+//   - message: Populated singular messages in src are merged into dst by
+//     recursively calling [xproto.PatchMessage].
+//   - list: The elements of every list field in src are appended to the
+//     corresponded list fields in dst.
+//   - map: The entries of every map field in src are MERGED (different from
+//     the behavior of proto.Merge) into the corresponding map field in dst.
+//
+// # Patch option "PATCH_REPLACE"
+//   - scalar: Same with "PATCH_MERGE".
```

Comment:
`Same with "PATCH_MERGE"` -> `Populated scalar fields in src are copied to dst.`